DOC-1871: RC - SAML Integration guide migration (#2345)
RedisLabs/redislabs-docs: content/rc/security/single-sign-on/saml-sso/saml-integration-okta-generic.md

5. The last step is an optional feedback step for Okta the **I'm an Okta customer adding an internal app** and click the **Finish** button.

    {{<image filename="images/rc/saml/okta_saml_app_int_6.png" alt="" >}}{{</image>}}

6. Next, scroll down the page of your newly created app integration and click on the **View Setup Instructions** button. This will open a new browser window and will give us the information needed to configure the IdP in Redis Cloud.

    {{<image filename="images/rc/saml/okta_saml_app_int_7.png" alt="" >}}{{</image>}}

   Scroll down to section 6 in the page, and note down the following information :
   
    * **Identity Provider Single Sign-On URL**
    * **Identity Provider Issuer**
    * **X.509 Certificate**

    {{<image filename="images/rc/saml/okta_saml_app_int_8.png" alt="" >}}{{</image>}}

   Once you have the information noted, you can close the window.

### Modify the application user profile

1. Click on "Directory" -> "Profile Editor" in the left hand menu, and choose the **Redis Cloud User**.

    {{<image filename="images/rc/saml/okta_saml_app_int_9.png" alt="" >}}{{</image>}}

> 2. We need to **add the custom attribute** to our user profile, this attribute will specify what Redis Cloud role the user has and on what SM account. Click on the **Add Attribute** button

Reviewer: "SM account" should be "Redis Cloud account".

    {{<image filename="images/rc/saml/okta_saml_app_int_10.png" alt="" >}}{{</image>}}

3. Fill in the following information for the new custom attribute.

    * Data type : **string array**
    * Display name : **redisAccountMapping**
    * Variable name : **redisAccountMapping**
    * Description : **redisAccountMapping**
    * Attribute required: **Yes**
    * Group priority : **Combine values across groups**

    {{<image filename="images/rc/saml/okta_saml_app_int_11.png" alt="" >}}{{</image>}}

4. Once the attribute has been added, it will appear in the list of attributes for the profile.

    {{<image filename="images/rc/saml/okta_saml_9.png" alt="" >}}{{</image>}}

## <a name="createtestuser">Step 2</a> - Create a group and assign the application

Now that our SAML IdP is configured, we need to create an OKTA group and assign the "Redis Cloud" application.

### Create the group

1. Click on "Directory" -> "Groups" in the left hand menu, and click on the **Add group** button.

    {{<image filename="images/rc/saml/okta_saml_group_1.png" alt="" >}}{{</image>}}

> 2. In the Add group popup, fill in the following information :

Reviewer: Maybe use "Enter" instead of "Fill in".

    {{<image filename="images/rc/saml/okta_saml_group_2.png" alt="" >}}{{</image>}}

    {{<image filename="images/rc/saml/okta_saml_group_3.png" alt="" >}}{{</image>}}

### Assign your users to the group

1. Click on the group and then click the "Assign people" button.

    {{<image filename="images/rc/saml/okta_saml_group_4.png" alt="" >}}{{</image>}}

2. For each user you wish to add to the group, highlight the user in the table and click on the **"+"** button. You can also use the "Add all" button if needed. Once you have added all the people to your group, click on the **"Save"** button.

    {{<image filename="images/rc/saml/okta_saml_group_5.png" alt="" >}}{{</image>}}

### Assign the application to the group

Now that your group is populated with its users, it is time to assign the SAML integration application to your group. From the menu "Applications" -> "Applications" -> "Redis Cloud" :

1. Click on the **Assign to groups** menu item.

    {{<image filename="images/rc/saml/okta_saml_group_6.png" alt="" >}}{{</image>}}

2. In the "Redis Cloud User Group" click on the "Assign" link.

    {{<image filename="images/rc/saml/okta_saml_group_7.png" alt="" >}}{{</image>}}

> 3. You will now define the redis account mapping string default for this group and click the **"Save and go back"** button. The key-value pair consists of the **lower-cased role name** (ie owner, member, manager, or viewer) AND your **Redis Cloud Account ID** (you can find this information in the upper-right user menu at app.redislabs.com). Finally, click on the **"Done"** button.

Reviewer 2: ```suggestion
3. You will now define the redis account mapping string default for this group and click **"Save and go back"**. The key-value pair consists of the **lower-cased role name** (ie owner, member, manager, or viewer) AND your **Redis Cloud Account ID** found in the [account settings]({{<relref "rc/accounts/account-settings">}}). Click **"Done"**.
```

    {{<image filename="images/rc/saml/okta_saml_group_8.png" alt="" >}}{{</image>}}

4. The mapping field has now been defined as a default for each member of the group.

    {{<image filename="images/rc/saml/okta_saml_group_9.png" alt="" >}}{{</image>}}

### Editing the mapping field for the group

> We might need to modify the Redis mapping field at a certain point in time. It is easy to do so by clicking on the "pencil" icon of the Redis Cloud group in the "Redis Cloud" application screen.

Reviewer 2: ```suggestion
To modify the Redis mapping field, select the "pencil" icon of the Redis Cloud group in the "Redis Cloud" application screen.
```

{{<image filename="images/rc/saml/okta_saml_group_10.png" alt="" >}}{{</image>}}

> An edit modal screen will appear, where you can modify the mapping field for the whole group.

Reviewer 2: ```suggestion
You can modify the mapping field for the whole group on the edit screen that appears.
```

{{<image filename="images/rc/saml/okta_saml_group_11.png" alt="" >}}{{</image>}}

### Editing the mapping field for a specific user

It is also possible to override the Redis mapping field at an individual user level. In order to do so, click on the "People" menu of the screen, and then on the "pencil" icon of the person you wish to modify the field for.

> {{<image filename="images/rc/saml/okta_saml_group_22.png" alt="" >}}{{</image>}}

Reviewer: @ggeorges2015 Broken image. I don't see this image's filename in the list of added files either.

> It is also possible to override the Redis mapping field at an individual user level. In order to do so, click on the "People" menu of the screen, and then on the "pencil" icon of the person you wish to modify the field for.

Reviewer 2: ```suggestion
To override the Redis mapping field at an individual user level, select the "People" menu of the screen, and then on the "pencil" icon of the person you wish to modify the field for.
```

Reviewer: Can you also change "and then on the pencil icon" to "and then select the pencil icon"? I would also generally recommend calling the "pencil icon" either "Edit", "Edit icon", or "Edit button" if that's what it is called in the UI and then include a small image of the pencil icon in the step.

> Next, we need to set the user's "Assignment master" to "Administrator". This will allow us to override the group's policy. We then click on the **"Save"** button.

Reviewer 2: ```suggestion
Set the user's "Assignment master" to "Administrator". This will allow us to override the group's policy. Select **"Save"**.
```

Reviewer: We also generally try to default to 2nd person pronouns instead of "we/us" where possible.

{{<image filename="images/rc/saml/okta_saml_group_13.png" alt="" >}}{{</image>}}

> We now see that the user's "Type" has been set to **"Individual"**

Reviewer 2: ```suggestion
The user's "Type" is set to **"Individual"**.
```

{{<image filename="images/rc/saml/okta_saml_group_14.png" alt="" >}}{{</image>}}

> In the final step, we can now click on the "pencil" icon of the user we wish to override their mapping. This will open a modal window which will allow you to modify the Redis mapping field.

Reviewer 2: ```suggestion
Select the "pencil" icon of the user to modify the Redis mapping field on the screen that appears.
```

{{<image filename="images/rc/saml/okta_saml_group_15.png" alt="" >}}{{</image>}}

{{<image filename="images/rc/saml/okta_saml_group_16.png" alt="" >}}{{</image>}}

## Step 3 - Configuring SAML support in Redis Cloud

Now that we have a test IdP server ready and our user group, we need to configure support for SAML in Redis Cloud.

### Sign in to your Redis Cloud SM account

Sign in to your SM account at [https://app.redislabs.com/#/login](https://app.redislabs.com/#/login)

### Activate SAML in Access Management

To activate SAML, you must have a local user (or social sign-on user) with the **owner** role. If you have the correct permissions, you will see the **Single Sign-On** tab.

1. Fill in the information you saved in [step 6 previously](#step6) in the **setup** form. This includes :

    * **Identity Provider Single Sign-On URL:**
    * **Identity Provider Issuer**
    * **X.509 Certificate**

    {{<image filename="images/rc/saml/sm_saml_1.png" alt="" >}}{{</image>}}

    {{<image filename="images/rc/saml/sm_saml_2.png" alt="" >}}{{</image>}}

> 2. Once you click the **enable** button, wait a few seconds for the status to change. You will then be able to **download** the service provider (SP) metadata. Save the file to your local hard disk.

Reviewer 2: ```suggestion
2. Once you select **Enable**, wait a few seconds for the status to change. You will then be able to **Download** the service provider (SP) metadata. Save the file to your local hard disk.
```

    {{<image filename="images/rc/saml/sm_saml_3.png" alt="" >}}{{</image>}}

3. Open the file in any text editor, and there are 2 pieces of information that you need to copy and mark down:

    * **EntityID** : The unique name of the service provider (SP)

>     {{<image filename="images/rc/saml/sm_sam1_4.png" alt="" >}}{{</image>}}

Reviewer: Broken image

> 3. Open the file in any text editor, and there are 2 pieces of information that you need to copy and mark down:

Reviewer 2: ```suggestion
3. Open the file in any text editor. Save the following text from the metadata:
```

    * **Location** : The location of the assertion consumer service

>     {{<image filename="images/rc/saml/sm_sam1_5.png" alt="" >}}{{</image>}}

Reviewer: Broken image

> 4. Return to you OKTA application, **click on Applications -> Redis Cloud -> General** and click the **Edit** link

Reviewer 2: ```suggestion
4. Return to OKTA, select **Applications > Redis Cloud > General** and select **Edit**.
```

    {{<image filename="images/rc/saml/okta_saml_app_int_12.png" alt="" >}}{{</image>}}

> 5. Then, navigate to the **Configure SAML** (step 2) and update the following information :

Reviewer 2: ```suggestion
5. Then, navigate to **Configure SAML** (step 2) and update the following information:
```

    * SAML Settings General

        * **Single sign on URL** : Use the information that you copied for **Location**
        * **Audience URI (SP Entity ID)** : Use the information that you copied for **EntityID**
        * **Default RelayState** : **Only needed if you want to have an IdP initiated flow**. Take the ID from the location URL in step 3 (the content after the last forward slash "/") and append to the url (ex: https://app.redislabs.com/#/login/?idpId=YOUR_LOCATION_ID).

    {{<image filename="images/rc/saml/okta_saml_app_int_13.png" alt="" >}}{{</image>}}

>     Once you have entered the information, click on the **Next** button, and finally click on the **Finish** button in feedback step.

Reviewer 2: ```suggestion
    Once you have entered the information, select **Next**. Finally. select **Finish**.
```

### Return to Redis Cloud SM

> 1. Return to Redis Cloud SM, and now click on the **Activate** button

Reviewer: "Redis Cloud SM" should be "Redis Cloud admin console"

    {{<image filename="images/rc/saml/sm_saml_8.png" alt="" >}}{{</image>}}

2. A popup will appear, explaining that in order to test the SAML connection, that we need to login with OKTA credentials of a user defined in the Redis Cloud group. This is a user who is part of the group that we assigned the Redis Cloud application to.

    {{<image filename="images/rc/saml/sm_saml_9.png" alt="" >}}{{</image>}}

3. The OKTA login screen will appear, enter the credentials and click **Sign In**

>     {{<image filename="images/rc/saml/sm_saml_14.png" alt="" >}}{{</image>}}

Reviewer: @ggeorges2015 It looks like this image is still broken. I don't see this image's filename in the list of added files, so maybe it was never checked in.

4. If the test has succeeded, you will see the following screen. Your local account is now considered a SAML account. In order to login to SM going forward, you click on the **Sign in with SSO** button.

    {{<image filename="images/rc/saml/sm_saml_11.png" alt="" >}}{{</image>}}

5. In the screen, enter your SAML email and click *Login*

    {{<image filename="images/rc/saml/okta_saml_app_int_15.png" alt="" >}}{{</image>}}

> 6. If there is only one user defined in the SM account, you will get the following popup window. Click on the **confirm** to convert the local user to a SAML user. 

Reviewer: "SM account" should be "Redis Cloud account"

> 6. If there is only one user defined in the SM account, you will get the following popup window. Click on the **confirm** to convert the local user to a SAML user. 

Reviewer: We also try to avoid the word "popup window" and usually use something like "dialog box" or "**Title** dialog".
